File: refactor_admin.py
import os

base_dir = r'c:\laragon\www\animal_php_myadmin\animal_php_myadmin\view\admin'
for root, dirs, files in os.walk(base_dir):
    for file in files:
        if file.endswith('.php'):
            path = os.path.join(root, file)
            with open(path, 'r', encoding='utf-8') as f:
                content = f.read()
            
            # Remove all require_once containing 'controller'
            lines = content.split('\n')
            new_lines = []
            has_auth = False
            for line in lines:
                if 'require_once' in line and 'controller' in line.lower():
                    continue
                if 'authorize' in line and 'ADMIN' in line:
                    has_auth = True
                new_lines.append(line)
            
            # Reconstruct content
            content = '\n'.join(new_lines)
            
            # If not authorized, add authorization at the top
            if not has_auth and '<?php' in content:
                # Find depth to config/env.php
                rel_path = os.path.relpath(path, base_dir)
                depth = rel_path.count(os.sep) + 2
                env_path = '/'.join(['..'] * depth) + '/config/env.php'
                
                auth_code = f"""
require_once __DIR__ . '/{env_path}';
$authController = new UserController();
$authController->authorize('ADMIN', '/Home');
"""
                content = content.replace('<?php', '<?php' + auth_code, 1)
                
            with open(path, 'w', encoding='utf-8') as f:
                f.write(content)

File: refactor_admin.php
<?php
$base_dir = __DIR__ . '/view/admin';

$iterator = new RecursiveIteratorIterator(new RecursiveDirectoryIterator($base_dir));
foreach ($iterator as $file) {
    if ($file->isFile() && $file->getExtension() === 'php') {
        $path = $file->getPathname();
        $content = file_get_contents($path);
        
        $lines = explode("\n", $content);
        $new_lines = [];
        $has_auth = false;
        
        foreach ($lines as $line) {
            // Remove require_once for controller
            if (stripos($line, 'require_once') !== false && stripos($line, 'controller') !== false) {
                continue;
            }
            if (strpos($line, 'authorize') !== false && strpos($line, 'ADMIN') !== false) {
                $has_auth = true;
            }
            $new_lines[] = $line;
        }
        
        $content = implode("\n", $new_lines);
        
        if (!$has_auth && strpos($content, '<?php') !== false) {
            // Calculate depth
            $rel_path = str_replace($base_dir . DIRECTORY_SEPARATOR, '', $path);
            $depth = substr_count($rel_path, DIRECTORY_SEPARATOR) + 2;
            $env_path = str_repeat('../', $depth) . 'config/env.php';
            
            $auth_code = "\nrequire_once __DIR__ . '/$env_path';\n";
            $auth_code .= "\$authController = new UserController();\n";
            $auth_code .= "\$authController->authorize('ADMIN', '/Home');\n";
            
            $content = preg_replace('/<\?php/', '<?php' . $auth_code, $content, 1);
        }
        
        file_put_contents($path, $content);
        echo "Processed $path\n";
    }
}
echo "Done.\n";
